### ISSUE TITLE
BUG: pd.SparseDtype('int', 0) == pd.SparseDtype('int', pd.NA) returns pd.NA

### ISSUE THREAD

**Issue by zfaee** [pandas-dev/pandas#54770]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

int_fill_0 = pd.SparseDtype('int', 0)
int_fill_n = pd.SparseDtype('int', pd.NA)

r = (int_fill_0 == int_fill_n)
# pd.NA

r_reverse = (int_fill_n == int_fill_0)
# False
```

### Issue Description

Equality comparison of SparseDtype filled by pd.NA and SparseDtype filled by non-na returns pd.NA, and is not symmetric.

### Expected Behavior

It should return False.

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

**Comment by kshitiz305** [comment]
take

**Comment by tqa236** [comment]
take

**Comment by joooeey** [comment]
> It should return False.

The [docs](https://pandas.pydata.org/docs/dev/user_guide/missing_data.html#propagation-in-arithmetic-and-comparison-operations) say that comparisons with `pd.NA` should evaulate to `pd.NA` (not to a boolean).

The fix #57783 implements NaN-like two-valued logic as exemplified by the tests: `assert a != b` and `assert b != a` where `b = SparseDtype(float, pd.NA)`. That is, after the fix, both comparisons in the OP evaluate to `False` but by the docs both should evaluate to `pd.NA`.
